spark-testing/spark_utils.py

```diff
@@ -127,10 +127,10 @@ def submit_job(
 
     conf_args = args.copy()
 
-    conf_args += [
-        '--conf spark.driver.memory=2g',
-        '--conf spark.mesos.role={}'.format(driver_role)
-    ]
+    conf_args += ['--conf spark.mesos.role={}'.format(driver_role)]
+
+    if "spark.driver.memory" not in conf_args:
```

Comment:
I don't think that this check makes sense. `conf_args` is a list, and contains, for example:
```python
[ ..., "--conf", "spark.driver.memory=20g", ...]
```
which means that this check will *always* be `False`. We would have to change the check to use something like:
```python
if not list(filter(lambda x: x.startswith("spark.driver.memory=""), conf_args)):
      conf_args += ['--conf spark.driver.memory=2g']
```

Reply:
dang.. good catch. I missed that one. I'll use your suggestion.